pandas/core/frame.py

```diff
@@ -1336,7 +1343,13 @@ def to_numpy(self, dtype=None, copy: bool = False) -> np.ndarray:
         array([[1, 3.0, Timestamp('2000-01-01 00:00:00')],
                [2, 4.5, Timestamp('2000-01-02 00:00:00')]], dtype=object)
         """
-        result = np.array(self.values, dtype=dtype, copy=copy)
+        result = self._mgr.as_array(
+            transpose=self._AXIS_REVERSED, dtype=dtype, na_value=na_value
+        )
+
+        if copy:
+            result = result.copy()
```

Comment:
It would be nice to only copy if `result` is actually not yet a copy. (but that might maybe easier to handle inside `as_array` as well?) Now this is not the most pressing issue